File: frontend/web/src/types/comment.ts
/**
 * Comment type definitions
 */

export interface Comment {
  id: string;
  article_id: string;
  parent_id: string | null;
  quote_id: string | null;
  author_id: string;
  author_name: string;
  content: string;
  is_deleted: boolean;
  edited: boolean;
  like_count: number;
  is_liked: boolean;
  version: number;
  created_at: string;
  updated_at: string;
}

/** Client-side tree node — same as Comment but with resolved children */
export interface CommentNode extends Comment {
  children: CommentNode[];
}

File: backend/modules/blogs/schemas/comment.py
"""
Comment Pydantic Schemas
"""
from datetime import datetime
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel, Field


# ── Requests ──────────────────────────────────────────────────────────────────

class CommentCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)
    parent_id: Optional[UUID] = None
    quote_id: Optional[UUID] = None


class CommentEdit(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)
    version: int = Field(..., ge=1)


# ── Response ──────────────────────────────────────────────────────────────────

class CommentResponse(BaseModel):
    """
    Flat comment row returned by the API.
    Frontend builds the tree from parent_id.

    When is_deleted=True, content/author fields are still returned as-is
    so the frontend can decide how to render the placeholder.
    """
    id: UUID
    article_id: UUID
    parent_id: Optional[UUID]
    quote_id: Optional[UUID]

    author_id: UUID
    author_name: str

    content: str
    is_deleted: bool
    edited: bool

    like_count: int
    is_liked: bool = False

    version: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CommentListResponse(BaseModel):
    items: List[CommentResponse]
    total: int
